src/app/rpmostree-compose-builtin-tree.c

```diff
@@ -475,52 +502,77 @@ install_packages_in_root (RpmOstreeTreeComposeContext  *self,
 
   if (generate_from_previous)
     {
-      if (!rpmostree_generate_passwd_from_previous (self->repo, rootfs_dfd,
+      const char *dest = opt_ex_unified_core ? "usr/etc/" : "etc/";
+      if (!rpmostree_generate_passwd_from_previous (self->repo, rootfs_dfd, dest,
                                                     treefile_dirpath,
                                                     self->previous_root, treedata,
                                                     cancellable, error))
         return FALSE;
     }
 
-  /* Before we install packages, drop a file to suppress the kernel.rpm dracut run.
-   * <https://github.com/systemd/systemd/pull/4174> */
-  const char *kernel_installd_path = "usr/lib/kernel/install.d";
-  if (!glnx_shutil_mkdir_p_at (rootfs_dfd, kernel_installd_path, 0755, cancellable, error))
-    return FALSE;
-  const char skip_kernel_install_data[] = "#!/usr/bin/sh\nexit 77\n";
-  const char *kernel_skip_path = glnx_strjoina (kernel_installd_path, "/00-rpmostree-skip.install");
-  if (!glnx_file_replace_contents_with_perms_at (rootfs_dfd, kernel_skip_path,
-                                                 (guint8*)skip_kernel_install_data,
-                                                 strlen (skip_kernel_install_data),
-                                                 0755, 0, 0,
-                                                 GLNX_FILE_REPLACE_NODATASYNC,
-                                                 cancellable, error))
-    return FALSE;
+  if (opt_ex_unified_core)
+    {
+      if (!rpmostree_context_import (self->corectx, cancellable, error))
+        return FALSE;
+      if (!rpmostree_context_relabel (self->corectx, cancellable, error))
```

Comment:
We don't need this relabel though, right? Otherwise IIUC, we will unnecessarily relabel the whole cache everytime the build env changes policy, even if the target tree didn't.

Reply:
It seems you're right...I am not entirely sure what I was thinking there.  Deleted in a fixup :arrow_down: